api.c

```diff
@@ -0,0 +1,1055 @@
+        {"date-str", required_argument, NULL, 511},
+        {"verif-text", required_argument, NULL, 512},
+        {"wrong-text", required_argument, NULL, 513},
+        {"keylayout", required_argument, NULL, 514},
+        {"noinput-text", required_argument, NULL, 515},
+        {"lock-text", required_argument, NULL, 516},
+        {"lockfailed-text", required_argument, NULL, 517},
+        {"greeter-text", required_argument, NULL, 518},
+        {"no-modkey-text", no_argument, NULL, 519},
+
+        // fonts
+        {"time-font", required_argument, NULL, 520},
+        {"date-font", required_argument, NULL, 521},
+        {"verif-font", required_argument, NULL, 522},
+        {"wrong-font", required_argument, NULL, 523},
+        {"layout-font", required_argument, NULL, 524},
+        {"greeter-font", required_argument, NULL, 525},
+
+        // text size
+        {"time-size", required_argument, NULL, 530},
+        {"date-size", required_argument, NULL, 531},
+        {"verif-size", required_argument, NULL, 532},
+        {"wrong-size", required_argument, NULL, 533},
+        {"layout-size", required_argument, NULL, 534},
+        {"modif-size", required_argument, NULL, 535},
+        {"greeter-size", required_argument, NULL, 536},
+
+        // text/indicator positioning
+        {"time-pos", required_argument, NULL, 540},
+        {"date-pos", required_argument, NULL, 541},
+        {"verif-pos", required_argument, NULL, 542},
+        {"wrong-pos", required_argument, NULL, 543},
+        {"layout-pos", required_argument, NULL, 544},
+        {"status-pos", required_argument, NULL, 545},
+        {"modif-pos", required_argument, NULL, 546},
+        {"ind-pos", required_argument, NULL, 547},
+        {"greeter-pos", required_argument, NULL, 548},
+
+        // text outline width
+        {"timeoutline-width", required_argument, NULL, 560},
+        {"dateoutline-width", required_argument, NULL, 561},
+        {"verifoutline-width", required_argument, NULL, 562},
+        {"wrongoutline-width", required_argument, NULL, 563},
+        {"modifieroutline-width", required_argument, NULL, 564},
+        {"layoutoutline-width", required_argument, NULL, 565},
+        {"greeteroutline-width", required_argument, NULL, 566},
+
+        // bar indicator stuff
+        {"bar-indicator", no_argument, NULL, 700},
+        {"bar-direction", required_argument, NULL, 701},
+        {"bar-orientation", required_argument, NULL, 703},
+        {"bar-step", required_argument, NULL, 704},
+        {"bar-max-height", required_argument, NULL, 705},
+        {"bar-base-width", required_argument, NULL, 706},
+        {"bar-color", required_argument, NULL, 707},
+        {"bar-periodic-step", required_argument, NULL, 708},
+        {"bar-pos", required_argument, NULL, 709},
+        {"bar-count", required_argument, NULL, 710},
+        {"bar-total-width", required_argument, NULL, 711},
+
+        {NULL, no_argument, NULL, 0}};
```

Comment:
Can you put the last closiung brace in its own line? Looks better that way, thanks.

Reply:
Looks better, but the same happens with [i3lock.c](https://github.com/Raymo111/i3lock-color/blob/305da376f578fc9d5cc89904b4fdcd68b69d89ea/i3lock.c#L1595-L1600) :D

Reply:
ok didn't realize this is copied code. Let's see if we can prevent thtat all together :P